packages/web-api/src/WebClient.ts

```diff
@@ -536,6 +550,23 @@ export class WebClient extends Methods {
   private buildResult(response: AxiosResponse): WebAPICallResult {
     let { data } = response;
 
+    // Check for GZIP response - if so, it is a successful response from admin.analytics.getFile
+    if (response.headers['content-type'] === 'application/gzip') {
+      // admin.analytics.getFile will return a Buffer that can be unzipped
+      try {
+        const unzippedData = zlib.unzipSync(data).toString().split('\n');
+        const fileData: string[] = [];
```

Comment:
Does this need to be a different type since we're pushing objects from `JSON.parse()` onto it? I tested accessing the output of this and I'm still able to access the objects normally (they aren't converted to strings and are successfully pushed into this array) but I'm wondering if there's a better convention for this?

I saw this convention as highlighted here (https://stackoverflow.com/questions/53964694/how-do-you-declare-an-array-of-objects-inside-typescript) for how to declare an array of objects, but it seems like it would be very involved and complex to declare this interface since we're expecting a large object for each line we're pushing into this array and also since I'm not sure if the output of each line of data from this API call is all expected to be the same structure of object. Would love to get thoughts on this!

Reply:
You can place an interface for the each row data under [the `response` directory](https://github.com/slackapi/node-slack-sdk/tree/main/packages/web-api/src/response). Perhaps, the name can be `AdminAnalyticsGetFileResponseDetails` or so. Also, the auto-generated [`AdminAnalyticsGetFileResponse.ts`](https://github.com/slackapi/node-slack-sdk/blob/%40slack/web-api%406.7.2/packages/web-api/src/response/AdminAnalyticsGetFileResponse.ts) does not have the file data property. Let's skip generating the interface from now on and modify it to have the details property. To prevent future modification by the code generator script, can you add if statement to check if the filename contains admin.analytics.getFile and skip the file generation for it here? https://github.com/slackapi/node-slack-sdk/blob/%40slack/web-api%406.7.2/scripts/code_generator.rb#L56

As for the interface details, if we don't have the bandwidth to work on the details right now, it's okay to go with any type as the first step. But if you want to complete the work this time, you can check the following resources:
* https://api.slack.com/methods/admin.analytics.getFile
* https://github.com/slackapi/java-slack-sdk/blob/v1.24.0/slack-api-client/src/main/java/com/slack/api/methods/response/admin/analytics/AdminAnalyticsGetFileResponse.java#L132-L476

The API document page for this endpoint is well-written. Also, you can refer to the Java SDK code to learn what properties are optional. This endpoint offers three response data patterns (1: public_channel, 2: member, 3: member+metadata_only). So, I think that there are three approaches to define the types:
1. Use unified interface covering all the patterns (most of the properties can be optional)
2. Define two types for the types (AdminAnalyticsGetFileResponseMemberDetails, AdminAnalyticsGetFileResponsePublicChannelDetails) and use a union type for the array
3. Define three types for covering all the possible patterns (~MemberDetails, ~PublicChannelMetadataOnlyDetails, ~PublicChannelDetails) and use a union type for the array

My suggestion here is either Option2 or Option3. Option2 should be great enough perhaps, but if we go with Option3, that'd be perfect for all use cases. Lastly, I know the suggested type names here are very long. I am fine to rename them if you would like to do so.